File: src/main/java/com/sup/dev/java_pc/sql/SqlQueryRemove.java
package com.sup.dev.java_pc.sql;

public class SqlQueryRemove extends SqlQueryWithWhere {

    private String table;

    public SqlQueryRemove(String table) {
        this.table = table;
    }

    @Override
    protected String createQuery() {
        return DELETE + table + getWhereString();
    }

}

File: src/main/java/com/sup/dev/java_pc/sql/SqlQueryRemove.kt
package com.sup.dev.java_pc.sql


class SqlQueryRemove(private val table: String) : SqlQueryWithWhere() {

    override fun createQuery(): String {
        return Sql.DELETE + table + createWhere()
    }


    override fun where(vararg wheres: SqlWhere.Where): SqlQueryRemove {
        return super.where(*wheres) as SqlQueryRemove
    }

    override fun where(columns: Any, condition: String, values: Any, link: String): SqlQueryRemove {
        return super.where(columns, condition, values, link) as SqlQueryRemove
    }
}
